cluster/images/karmada-agent/Dockerfile

```diff
@@ -1,14 +1,10 @@
-FROM golang:1.17-alpine3.15 AS builder
```

Comment:
just question:
What's the difference between @golang:1.17-alpine3.15 and golang:1.17 ? Why do you think `golang:1.17` is better?

Reply:
1: whatever golang:1.17-alpine3.15 or golang:1.17 ,they are just build environment which were not put into the final image
2: while go1.17 can execute make comand but alpine can not